src/precompile_utils.c

```diff
@@ -374,9 +374,8 @@ static void *jl_precompile_trimmed(size_t world)
     jl_value_t *ccallable = NULL;
     JL_GC_PUSH2(&m, &ccallable);
     jl_method_instance_t *mi;
-    while (1)
-    {
-        mi = (jl_method_instance_t*)arraylist_pop(jl_entrypoint_mis);
+    for (size_t i = 0; i < jl_entrypoint_mis->len ; i++) {
+        mi = (jl_method_instance_t*)jl_entrypoint_mis->items[i];
```

Comment:
One worry I have here is that there is the potential to break user-code that depends on this behavior.
That code may be rare, but I have seen it in the wild.

Reply:
I think we should stick to default-exclude for everything with trimming; it will keep it easier to use.